flocker/testtools/test/test_base.py

```diff
@@ -0,0 +1,71 @@
+"""
+Tests for flocker base test cases.
+"""
+
+import errno
+import shutil
+
+from hypothesis import given
+from hypothesis.strategies import binary
+from testtools import TestCase
+from testtools.matchers import (
+    DirExists,
+    Not,
+    PathExists,
+)
+from testtools.testresult.doubles import Python27TestResult
+from twisted.python.filepath import FilePath
+from twisted.trial import unittest
+
+from .._base import AsyncTestCase
+
+
+class AsyncTestCaseTests(TestCase):
+
+    @given(binary())
```

Comment:
> If the test ever fails and I have to read the (one list) != (another list) output including the generated value, does this mean I might end up looking at 80kB of binary junk?

No, because hypothesis has awesome shrinking logic. If it finds a failing case, it'll do its level best to find a smaller failing case. 

> Would configuring a smallish average here be a reasonable use of Hypothesis? Something around 30, say.

Yes. We could also restrict the alphabet, but I think it's best to wait until it's a problem.